src/Http/Routing/src/MapAction/MapActionEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.MapAction;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="IEndpointRouteBuilder"/> to define HTTP API endpoints.
+    /// </summary>
+    public static class MapActionEndpointRouteBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches the pattern specified via attributes.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="action">The delegate executed when the endpoint is matched.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static IEndpointConventionBuilder MapAction(
```

Comment:
Consider returning an implementation of IEndpointConventionBuilder for this API. We do this for MVC, Razor pages, SignalR, gRPC.

e.g. https://github.com/dotnet/aspnetcore/blob/f9c20d290b053073baeffb949ae10a0782beeef1/src/SignalR/server/SignalR/src/HubEndpointRouteBuilderExtensions.cs#L27

I don't think anything uses these specific types. It's just in case so the ability is there.